staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto

```diff
@@ -1342,3 +1347,56 @@ message ReopenContainerLogRequest {
 
 message ReopenContainerLogResponse{
 }
+
+// Common options used for checkpointing and restoring.
+message CheckpointRestoreOptions {
+    // Keep temporary files. Like log files. Helpful for debugging.
+    bool keep = 1;
+    // Checkpoint/Restore the container with established TCP connections.
+    bool tcp_established = 2;
```

Comment:
is it possible to checkpoint and restore with established tcp connections ? How does that work ?

Reply:
> is it possible to checkpoint and restore with established tcp connections ?

Yes it is possible to checkpoint and restore established TCP connections. On the lower levels, checkpointing single processes or containers using runc/Podman it makes more sense when talking about the TCP connection from an outside client to the process or container. The restored process or container needs to have access to the same IP or the restore will fail.

In the Kubernetes context, especially for the current `drain --checkpoint` implementation, it might seem less useful for connections to external clients. It is, however, important to have the possibility to checkpoint and restore established TCP connections for containers and Pods which have open TCP connection between each other.

> How does that work ?

Please have a look at how CRIU does that: https://criu.org/TCP_connection